File: mpp/lab3/src/main/java/service/Service.java
package service;

import domain.Arbiter;
import domain.Participant;
import domain.Score;
import domain.ScoreType;
import observer.Observable;
import repository.IArbiterRepository;
import repository.IParticipantRepository;
import repository.IScoreRepository;
import repository.RepositoryError;

import java.util.List;

public class Service extends Observable {
    public static final String SCORE_SET_CHANGE = "score_set_change";

    private final IParticipantRepository participantRepository;
    private final IArbiterRepository arbiterRepository;
    private final IScoreRepository scoreRepository;

    public Service(IParticipantRepository participantRepository, IArbiterRepository arbiterRepository, IScoreRepository scoreRepository) {
        this.participantRepository = participantRepository;
        this.arbiterRepository = arbiterRepository;
        this.scoreRepository = scoreRepository;
    }

    public Arbiter loginArbiter(String name, String password) throws RepositoryError {
        return arbiterRepository.findByNameAndPassword(name, password);
    }

    public void setScore(Score score, Arbiter arbiter) throws RepositoryError {
        scoreRepository.setScore(score.getParticipant().getId(), arbiter.getType(), score.getScore(arbiter.getType()));
        change(SCORE_SET_CHANGE, null, score);
    }

    public List<Score> getScoresForType(ScoreType type) throws RepositoryError {
        return scoreRepository.findScoresForTypeSortedDescending(type);
    }

    public List<Score> getScores() throws RepositoryError {
        return scoreRepository.findScoresSortedByName();
    }
}

File: mpp/lab7/server/Service/Service.cs
using System.Collections.Generic;
using server.Domain;
using server.Repository;

namespace server.Service
{
    public class Service : BaseService
    {
        private IParticipantRepository ParticipantRepository { get; }
        private IArbiterRepository ArbiterRepository { get; }
        private IScoreRepository ScoreRepository { get; }

        public Service(IParticipantRepository participantRepository,
            IArbiterRepository arbiterRepository,
            IScoreRepository scoreRepository)
        {
            ParticipantRepository = participantRepository;
            ArbiterRepository = arbiterRepository;
            ScoreRepository = scoreRepository;
        }
        
        public Arbiter LoginArbiter(string name, string password) {
            return ArbiterRepository.FindByNameAndPassword(name, password);
        }

        public Score SetScoreValue(int participantId, ScoreType type, int value) {
            ScoreRepository.Set(participantId, type, value);
            var score = ScoreRepository.Get(participantId);
            ObserverSetScore(score);
            return score;
        }

        public List<Score> GetScores() {
            return ScoreRepository.FindSortedByName();
        }
        
        public List<Score> GetRankings(ScoreType type) {
            return ScoreRepository.FindByTypeSortedDescending(type);
        }
    }
}